### ISSUE TITLE
Can you use python to train a network from scratch?

### ISSUE THREAD

**Issue by zergylord** [BVLC/caffe#360]
I'm trying to train a model completely in python (the training process is interactive enough to justify this I think), but the only way I've seen to get a network into python is by calling:

caffe.Net(model_def_file, pretrained_model)

But this presumes you already have a model to work with. Is there a way to create a new model in python? Or if not, is there an easy way to create a model file without training it on anything (to use it as the pretrained_model argument)?

I hope this isn't a silly question; I've tried figuring it out myself, but the document for the python wrapper is pretty sparse.

**Comment by erogol** [comment]
How should I feed the data to SGDsolver in Python interface

**Comment by bhack** [comment]
@erogol Use the mailing list for support question

**Comment by reddusk** [comment]
Hi @zergylord @longjon . I am working on semantic segmentation following the paper by @longjon and @shelhamer . I have already configured the prototxt files that you have provided for my needs and it is working as intended. However, the mean accuracy fluctuates between 42% to 48% while overall accuracy is around 86%. Also, I feel that the network is very deep for my purposes and hence takes a lot of time to train. Therefore, I want to train my own network using caffe's  python interface . My dataset is simple with 4 different categories but in each image it is guaranteed that it will have white/grey background and a single object belonging to one of the 4 categories. I read the above posts and modified my prototxt to use the memory data layer. Below is the first snippet of my data layer as advised by @longjon :
`layer
{
name: "data"
type: "MemoryData"
top: "data"
top: "label"
memory_data_param {
    batch_size: 100
    channels: 3
    height: 300
    width: 300
  }
}`
All this works fine but the net takes my label shape as 100 while in semantic segmentation I need a matrix of ground truth labels. So my labels are of dimension(300x300) and in the python code I convert them to 4D. It throws a shape mismatch error at the final loss layer. How can I indicate in the prototxt files that my labels are actually 4D arrays as required by `solver.net.set_input_arrays(data, labels)` ? Or is there any other way to train my own network for semantic segmentation?
